shared/actions/chat.js

```diff
@@ -791,15 +796,7 @@ function * _incomingMessage (action: IncomingMessage): SagaGenerator<any, any> {
       const setStatus: ?SetStatusInfo = action.payload.activity.setStatus
       if (setStatus) {
         yield call(_updateInbox, setStatus.conv)
-        const conversationIDKey = conversationIDToKey(setStatus.convID)
-        const muted = setStatus.status === CommonConversationStatus.muted
-        yield put(({
-          payload: {
-            conversationIDKey,
-            muted,
-          },
-          type: 'chat:conversationSetStatus',
-        }: ConversationSetStatus))
+        yield call(_ensureValidSelectedChat, false, true)
```

Comment:
No, not quite.  _ensureValidSelectedChat gets called from more places than just here, and we wouldn't want it to deselect in those other places.  Also, even at this call site, we get here if e.g. a conversation is muted, and when that happens you wouldn't want to deselect.